pkg/bgpv1/manager/reconcile.go

```diff
@@ -198,7 +200,37 @@ func (r *NeighborReconciler) Priority() int {
 }
 
 func (r *NeighborReconciler) Reconcile(ctx context.Context, params ReconcileParams) error {
-	return neighborReconciler(ctx, params.Server, params.NewC, params.CState)
+	return neighborReconciler(ctx, params.Server, params.NewC, params.CState, params.SecretFetcher)
+}
+
+func fetchPeerPassword(sc *ServerWithConfig, n *v2alpha1api.CiliumBGPNeighbor, secretFetcher types.SecretFetcher) (string, bool, error) {
+	l := log.WithFields(
+		logrus.Fields{
+			"component": "manager.fetchPeerPassword",
+		},
+	)
+	if n.AuthSecretRef != "" {
+		secret, err := secretFetcher.Fetch(n.AuthSecretRef)
+		if err != nil {
+			return "", true, fmt.Errorf("failed to fetch secret %q: %w", n.AuthSecretRef, err)
+		}
+		tcpPassword := string(secret["password"])
+		if tcpPassword == "" {
+			return "", true, fmt.Errorf("secret %q: missing password key", n.AuthSecretRef)
+		}
+
+		key := fmt.Sprintf("%s%d", n.PeerAddress, n.PeerASN)
+		old := sc.PeerPasswords[key]
+		if old != tcpPassword {
+			l.Debugf("Updated TCP password from secret %q", n.AuthSecretRef)
+			sc.PeerPasswords[key] = tcpPassword
```

Comment:
I'd prefer separating secret fetch and comparing peer passwords into different functions. Also, changing the server config state inside the function named `fetch` feel a bit odd to me. If we have such a side-effect, then let's change the function name to `update` or something.